### PR TITLE
substrate runner: increase line read + dump CLI output if parsing fails

### PR DESCRIPTION
**Pull request by niklasad1**

### CODE REVIEW

File: testing/substrate-runner/src/lib.rs [hunk 1]
```diff
@@ -246,14 +247,18 @@ impl Drop for SubstrateNode {
 // locate the port number that is logged out to it.
 fn try_find_substrate_port_from_output(
     r: impl Read + Send + 'static,
-) -> (Option<u16>, Option<String>, Option<u32>) {
+) -> (Option<u16>, Option<String>, Option<u32>, String) {
```
Comment (jsdw) [inline testing/substrate-runner/src/lib.rs:250]: Nit: probably itme to make this a named struct :D

**Review by jsdw — APPROVED** [review]
Nice: hopefully that'll make it easier to debug failures!

**Review by lexnv — APPROVED** [review]

File: testing/substrate-runner/src/lib.rs [hunk 2]
```diff
@@ -246,14 +247,18 @@ impl Drop for SubstrateNode {
...
     let mut port = None;
     let mut p2p_address = None;
     let mut p2p_port = None;
 
+    let mut log = String::new();
+
     for line in BufReader::new(r).lines().take(50) {
```
Comment (niklasad1) [inline testing/substrate-runner/src/lib.rs:257]:
```suggestion
    for line in BufReader::new(r).lines().take(100) {
```

File: testing/integration-tests/src/full_client/client/unstable_rpcs.rs [hunk 3]
```diff
@@ -353,6 +359,8 @@ async fn transaction_v1_broadcast() {
 
     while let Some(finalized) = finalized_sub.next().await {
         let finalized = finalized.unwrap();
+        tracing::info!("Finalized block: {:?}", finalized.hash());
```
Comment (niklasad1) [inline testing/integration-tests/src/full_client/client/unstable_rpcs.rs:362]:
```suggestion
```

File: testing/integration-tests/src/full_client/client/unstable_rpcs.rs [hunk 4]
```diff
@@ -353,6 +359,7 @@ async fn transaction_v1_broadcast() {
 
     while let Some(finalized) = finalized_sub.next().await {
         let finalized = finalized.unwrap();
+        tracing::info!("Number of blocks left: {}", num_blocks);
```
Comment (niklasad1) [inline testing/integration-tests/src/full_client/client/unstable_rpcs.rs:362]:
```suggestion
```

File: testing/substrate-runner/src/lib.rs [hunk 5]
```diff
@@ -301,7 +304,43 @@ fn try_find_substrate_port_from_output(
                 .unwrap_or_else(|_| panic!("valid port expected for log line, got '{port_str}'"));
             p2p_port = Some(port_num);
         }
+
+        if port.is_some() && p2p_address.is_some() && p2p_port.is_some() {
```
[the remark this answers is not in the document]
Comment (jsdw) [reply testing/substrate-runner/src/lib.rs:308]: ooh good catch!

**Review by jsdw — APPROVED** [review]

**Pull request merged**